pycheribuild/projects/repository.py

```diff
@@ -176,6 +176,7 @@ def __init__(
         force_branch: bool = False,
         temporary_url_override: "Optional[str]" = None,
         url_override_reason: "typing.Any" = None,
+        temporary_branch_override: "Optional[str]" = None,
```

Comment:
I'm not convinced temporary_url_override was a great idea rather than a comment, and I'm even less convinced temporary_branch_override is worth introducing versus just changing default_branch and commenting a TODO/XXX/whatever with the why and what it should become

Reply:
The url_override does check if the current remote is one of the old_urls and prompts to update, so there is some (minimal) value there. But I agree with the branch override there is no actual checking that the current branch upstream is stale, so probably better off with a comment. Unlike the url_override it only affects cloning, so a comment should be fine.